src/mrack/providers/aws.py

```diff
@@ -209,9 +209,41 @@ async def validate_hosts(self, reqs):
         """Validate that host requirements are well specified."""
         return
 
+    def get_subnet_available_ips(self, subnet_id, log_msg_start):
+        """Get number of IPs available in a subnet."""
+        try:
+            subnet = self.ec2.Subnet(subnet_id)
+        except ClientError:
+            err_msg = (
+                f"{log_msg_start} Error retrieving " f"info from subnet: {subnet_id}"
+            )
+            logger.error(err_msg)
+            raise
+
+        return subnet.available_ip_address_count
+
     async def can_provision(self, hosts):
```

Comment:
One observation about this method:
It is not part of this PR, but it is possible also to check CPU limits (like we do on Openstack).
e.g. vCpus limit number rigth now is 640.
Ref: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-on-demand-instances.html#ec2-on-demand-instances-limits  

This is not limiting rigth now, but just in case, it could be useful in the future.